Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1796 @@
+
+chmod +x /etc/init.d/solr
+```
+Insert the following in `/etc/init.d/solr` at line 17 (just after the `chkconfig` block):
+```
+# chkconfig: 2345 64 36
+# description: Jetty9 with Solr
+### BEGIN INIT INFO
+# Provides: solr
+# Required-Start: $all
+# Required-Stop: $all
+# Default-Start: 2 3 4 5
+# Default-Stop: 0 1 6
+# Short-Description: Start solr/jetty9 at boot.
+# Description: Start the jetty9 servelet container holding Solr.
+### END INIT INFO
+```
+Note: We are putting Solr data under `/srv/`, as with Fedora data, as part of the move away from "everything under Fedora".
+```
+mkdir -p /srv/solr/data
+
+chown -R jetty:jetty /srv/solr
+
+cat > /etc/systemd/system/solr.service<<END_SS
+[Unit]
+Description=Solr Index running under jetty9
+SourcePath=/etc/init.d/solr
+
+[Service]
+Type=simple
+ExecStart=/etc/init.d/solr start
+ExecStop=/etc/init.d/solr stop
+#Restart=/etc/init.d/solr restart
+
+[Install]
+WantedBy=multi-user.target
+END_SS
+
+systemctl enable solr.service
+```
+Prepare `fedoragsearch`:
+```
+cd $CATALINA_HOME/webapps/fedoragsearch/FgsConfig
+
+ant generateIndexingXslt
+
+### NOTE: Choose one of these, first for Solr under Fedora Tomcat, second for stand alone Solr
+#ant -f fgsconfig-basic.xml -Dlocal.FEDORA_HOME=$FEDORA_HOME -DgsearchUser=$FEDORA_ADMIN_USER -DgsearchPass=$FEDORA_ADMIN_PASS -DfinalConfigPath=$CATALINA_HOME/webapps/fedoragsearch/WEB-INF/classes -DlogFilePath=$FEDORA_HOME/server/logs -DfedoraUser=$FEDORA_ADMIN_USER -DfedoraPass=$FEDORA_ADMIN_PASS -DobjectStoreBase=$FEDORA_HOME/data/objectStore -DindexDir=$FEDORA_HOME/solr/$SOLR_DEFAULT_CORE_PATH/data/index -DindexingDocXslt=foxmlToSolr -propertyfile fgsconfig-basic-for-islandora.properties
+
+#ant -f fgsconfig-basic.xml -Dlocal.FEDORA_HOME=$FEDORA_HOME -DgsearchUser=$FEDORA_ADMIN_USER -DgsearchPass=$FEDORA_ADMIN_PASS -DfinalConfigPath=$CATALINA_HOME/webapps/fedoragsearch/WEB-INF/classes -DindexEngine=Solr -DindexBase=http://127.0.0.1:8983/solr/collection1 -DlogFilePath=$FEDORA_HOME/server/logs -DfedoraUser=$FEDORA_ADMIN_USER -DfedoraPass=$FEDORA_ADMIN_PASS -DfedoraVersion=$FEDORA_VERSION -DobjectStoreBase=/srv/data/objectStore -DindexDir=/srv/solr/data/index -DindexingDocXslt=foxmlToSolr -propertyfile fgsconfig-basic-for-islandora.properties
+
+### backup some default files which will be clobbered in the next section.
+cd $SOLR_HOME/solr/$SOLR_DEFAULT_CORE_PATH/conf
+
+mv schema.xml schema.xml.bak  
+
+mv solrconfig.xml solrconfig.xml.bak  
+
+cd $CATALINA_HOME/webapps/fedoragsearch/WEB-INF/classes/fgsconfigFinal/index/FgsIndex/
+
+mv foxmlToSolr.xslt foxmlToSolr.xslt.bak 
+```
+
+##### Solr Configuration <a id="solr-configuration"></a>
+```
+cd ~
+
+git clone --recursive git://github.com/discoverygarden/basic-solr-config.git 
+
+cd basic-solr-config 
+
+### NOTE: Choose one of the following two, the first if you are using Solr 4.2.0, the second if you are using Solr 4.10.4.
+#git checkout 4.x
+
+#git checkout 4.10.x
```

Comment:
https://github.com/discoverygarden/basic-solr-config/blob/4.10.x/README.md

"You also have to copy the jars from solr/example/lib/ext/* to $CATALINA_HOME/webapps/solr/WEB-INF/lib/ for more information see https://wiki.apache.org/solr/SolrLogging#Using_the_example_logging_setup_in_containers_other_than_Jetty"

Reply:
This is accomplished by the DigiBESS procedure, which copies the entire 'example'
directory as the Solr home.

Reply:
Ah I see that now. Thanks!